src/common/aggregate/aggregator.js

```diff
@@ -10,6 +10,10 @@ export class Aggregator extends SharedContext {
   constructor (parent) {
     super(parent)
     this.aggregatedData = {}
+
+    /* Unfortunately, there's no other good medium through which SR can signal it's ready to other features that depends on it.
+      Here, the central aggregator shared across all feature instrument & aggregate components is used for that signal. */
+    this.sessionReplayInitialized = new Promise(resolve => this.onReplayReady = resolve)
```

Comment:
This still feels like a really weird place to put this.  I think the shared aggregator can be stripped out as (most) of the features do not utilize it, so that would make this a casualty of that effort

Reply:
If the shared promise thing is the way you want to do this instead of using the event emitter, we should really have some other layer defined in the `agent.js` that handles these sort of cross-feature links.

Reply:
You mean move this to Trace instrument and add a handler subscribed to a new event emitted by Replay aggregate after it starts recording, and pass this promise from Trace instrument -> aggregate -> `getSessionReplayMode` ?

Reply:
Promise can also live in `replay-mode.js` but the handler tying event to resolve needs to happen pre-load. So will need to add a stateful init to the module so that it sets up the handler & promise with the instruments.

Reply:
I agree this probably shouldn't be in the aggregator class. It sounds like we can't use the event emitter because we can't guarantee the load order and that the subscribe will be in place when the event is dispatched indicating the session replay feature loaded. Is that correct? If so, I wonder if we could take this as a need to enhance the event emitter to allow for replaying of events for certain streams? Or, maybe the base feature class can have an `isInitialized` deferred object that can be resolved? Something along the lines of setting up lifecycle events for features. I digress, these would all be later PRs and work.

Reply:
We settled on creating a new `sharedXYZ` which will basically be a side-by-side obj to the sharedAggregator. Just use it for whatever inter-feature comm needs to happen.